src/main/scala/mimir/algebra/function/IntervalFunctions.scala

```diff
@@ -0,0 +1,144 @@
+package mimir.algebra.function;
+
+import mimir.algebra._
+import org.joda.time.DateTime
+import org.joda.time.Period;
+
+object IntervalFunctions
+{
+
+  def register(fr: FunctionRegistry)
+  {
+
+
+
+    fr.register(
+      "INTERVALPLUSINTERVAL",
+      (x) => {
+	var p = x(0).asInterval.plus(x(1).asInterval);
+	IntervalPrimitive(p.getYears,p.getMonths,p.getWeeks,p.getDays,p.getHours,p.getMinutes,p.getSeconds,p.getMillis)
+      },
+      (_) => TInt()
+    )
+
+
+    fr.register(
+      "INTERVALSUBINTERVAL",
+      (x) => {
+	var p = x(0).asInterval.minus(x(1).asInterval);
+	IntervalPrimitive(p.getYears,p.getMonths,p.getWeeks,p.getDays,p.getHours,p.getMinutes,p.getSeconds,p.getMillis)
+      },
+      (_) => TInt()
+    )
+
+    fr.register(
+      "DATEPLUSINTERVAL",
+      (x) => {
+	val d = x(0).asDateTime.plus(x(1).asInterval)
+        TimestampPrimitive(d.getYear,d.getMonthOfYear,d.getDayOfMonth,d.getHourOfDay,d.getMinuteOfHour,d.getSecondOfMinute,d. getMillisOfSecond)
+      },
+      (_) => TInt()
+    )
+
+    fr.register(
+      "DATESUBINTERVAL",
+      (x) => {
+	val d = x(0).asDateTime.minus(x(1).asInterval)
+        TimestampPrimitive(d.getYear,d.getMonthOfYear,d.getDayOfMonth,d.getHourOfDay,d.getMinuteOfHour,d.getSecondOfMinute,d. getMillisOfSecond)
+      },
+      (_) => TInt()
+    )
+
+
+    fr.register(
+      "DATESUBDATE",
+      (x) => {
+	val p = new Period(x(1).asDateTime,x(0).asDateTime)
+        IntervalPrimitive(p.getYears,p.getMonths,p.getWeeks,p.getDays,p.getHours,p.getMinutes,p.getSeconds,p.getMillis)
+      },
+      (_) => TInt()
+    )
+
```

Comment:
Operators subsumed by Arithmetic aren't necessary.

It might be handy to keep the extraction functions (MILLISECONDS OF, etc...).